src/analysis/cauchy_equation.lean

```diff
@@ -0,0 +1,156 @@
+/-
+Copyright (c) 2022 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+import analysis.normed_space.pointwise
+import measure_theory.measure.haar_lebesgue
+
+/-!
+# Cauchy's Functional Equation
+
+This file contains the classical results about the Cauchy's functional equation
+`f (x + y) = f x + f y` for functions `f : ℝ → ℝ`. In this file, we prove that the solutions to this
+equation are linear up to the case when `f` is a Lebesgue measurable functions, while also deducing
+intermediate well-known variants.
+-/
+
+open add_monoid_hom measure_theory measure_theory.measure metric nnreal set
+open_locale pointwise topological_space
+
+variables {ι : Type*} [fintype ι] {s : set ℝ} {a : ℝ}
+
+local notation `ℝⁿ` := ι → ℝ
+
+/-- **Cauchy's functional equation**. An additive monoid homomorphism automatically preserves `ℚ`.
+See also `add_monoid_hom.to_rat_linear_map`. -/
+lemma add_monoid_hom.is_linear_map_rat (f : ℝ →+ ℝ) : is_linear_map ℚ f :=
+⟨map_add f, map_rat_cast_smul f ℝ ℝ⟩
```

Comment:
If you formulate it in the `add_monoid_hom` namespace, then this should be stated for any real TVS and should located right next to `add_monoid_hom.to_rat_linear_map`.